### PR TITLE
Add the interface of intel endpoint

### PR DESCRIPTION
**Pull request by sungmanch**
Now, we can see the results with the OpenVINO Model.
image

image

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/TBD-Labs-AI/latched/blob/repo-setting/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/latched/endpoints/base.py [hunk 1]
```diff
@@ -1,12 +1,34 @@
 # Copyright 2024 TBD Labs Inc. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-from latched.model_wrappers.base import BaseModelWrapper
+from __future__ import annotations
```
Comment (psycoder-sup) [inline src/latched/endpoints/base.py:5]: What does this code do?

File: src/latched/endpoints/intel.py [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+    ):
+        super().__init__(model_wrapper, tokenizer, model_path, task)
+        self.ov_model = self._prepare_model()
+        self.pipeline = self._prepare_pipeline()
+
+    def _prepare_model(self) -> nn.Module:
+        if self.model_wrapper.library_name == "transformers":
+            from optimum.intel import OVModelForCausalLM
+
+            return OVModelForCausalLM.from_pretrained(self.model_path)
```
Comment (psycoder-sup) [inline src/latched/endpoints/intel.py:50]: if a user brings their custom model, does this work?

**Review by psycoder-sup — COMMENTED** [review]
Please check the added comments

File: src/latched/endpoints/intel.py (hunk 2 above)
Comment (sungmanch) [reply src/latched/endpoints/intel.py:50]: Not work, current code can't consider custom cases and other framework also. need to consider the more general design later.

File: src/latched/endpoints/base.py (hunk 1 above)
Comment (sungmanch) [reply src/latched/endpoints/base.py:5]: If this code line isn't included, `mypy` returns the error for the `TYPE_CHECKING`.

**Review by psycoder-sup — APPROVED** [review]

**Pull request merged**